src/iris/api.py

```diff
@@ -4653,6 +4684,437 @@ def on_get(self, req, resp, incident_id):
         conn.close()
 
 
+class NotificationCategories(object):
+    allow_read_no_auth = True
+
+    def on_get(self, req, resp):
```

Comment:
A use case we're probably going to employ in the front end is getting all the applications that have custom categories defined to populate the application dropdown/typeahead that users can add custom delivery preferences for. There probably isn't going to be a lot of applications with custom categories defined but it might be worth it to be able to return just a list of applications either through filtering or its own endpoint.

Reply:
This can be handled via the existing Applications GET endpoint, with client-side filtering. We might want to implement server-side filters at some point though, as we get more applications. On the other hand, the number of apps is always going to be fairly limited.